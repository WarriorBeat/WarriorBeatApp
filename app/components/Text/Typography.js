/**
 * Typography.js
 * Typography Components
 * components/Text
 */

import React from "react"
import PropTypes from "prop-types"
import HTML from "react-native-render-html"
import { StyleSheet } from "react-native"
import { Colors } from "components/styles"
import Text from "./Text"
import * as TextStyle from "./styles"

const styles = StyleSheet.create({
  html: {
    ...TextStyle.type.body,
    ...Colors.color("blackPrimary"),
    fontSize: 16,
  },
})

export const Header = ({ children, Color, Content }) => (
  <Text Weight="semibold" Color={Color}>
    {Content || children}
  </Text>
)

export const Title = ({ children, Color, Content }) => (
  <Text Weight="semibold" Type="title" Color={Color}>
    {Content || children}
  </Text>
)

export const Body = ({ Color, children, Content }) => (
  <Text fontSize={16} Color={Color} Type="body">
    {Content || children}
  </Text>
)

export const HTMLBody = ({ Content }) => <HTML html={Content} baseFontStyle={styles.html} />

Header.propTypes = {
  children: PropTypes.node.isRequired,
  Color: PropTypes.string,
  Content: PropTypes.string,
}

Header.defaultProps = {
  Color: "black",
  Content: null,
}

Title.propTypes = Header.propTypes
Title.defaultProps = Header.defaultProps

Body.propTypes = Header.propTypes
Body.defaultProps = {
  Color: "blackPrimary",
  Content: null,
}

HTMLBody.propTypes = {
  Content: HTML.propTypes.html.isRequired,
}
