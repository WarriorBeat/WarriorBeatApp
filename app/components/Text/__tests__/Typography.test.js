/**
 * __tests__/Typography.test.js
 * Component Test
 * tests
 */
/* eslint-disable */

import React from "react"
import { render } from "react-native-testing-library"
import * as Typo from "../Typography"
import { View } from "react-native"

test("should render correctly", () => {
  const tree = render(
    <View>
      <Typo.Header>Hello!</Typo.Header>
      <Typo.Body Content="Hello!" />
    </View>,
  )
  expect(tree).toMatchSnapshot()
})

test("should render HTML correctly", () => {
  const tree = render(<Typo.HTMLBody Content="<p>Hello!</p>" />)
  expect(tree).toMatchSnapshot()
})
