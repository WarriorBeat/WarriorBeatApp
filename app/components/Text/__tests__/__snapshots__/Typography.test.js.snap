// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HTML correctly 1`] = `
<View
  style={Object {}}
>
  <Text
    allowFontScaling={true}
    selectable={false}
    style={
      Array [
        Object {
          "marginBottom": 16,
          "marginTop": 16,
        },
        Object {},
        Object {},
      ]
    }
  >
    <Text
      allowFontScaling={true}
      selectable={false}
      style={
        Array [
          Object {},
          Object {},
        ]
      }
    >
      <Text
        allowFontScaling={true}
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "rgba(0,0,0,0.87)",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "400",
            "letterSpacing": -0.408,
            "lineHeight": 22,
            "marginBottom": 16,
            "marginTop": 16,
          }
        }
      >
        Hello!
      </Text>
    </Text>
  </Text>
</View>
`;

exports[`should render correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#000",
        "fontFamily": "System",
        "fontSize": undefined,
        "fontStyle": "normal",
        "fontWeight": "600",
        "includeFontPadding": true,
        "textTransform": undefined,
      }
    }
  >
    Hello!
  </Text>
  <Text
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "rgba(0,0,0,0.87)",
        "fontFamily": "System",
        "fontSize": 16,
        "fontStyle": "normal",
        "fontWeight": "400",
        "includeFontPadding": true,
        "letterSpacing": -0.408,
        "lineHeight": 22,
        "textTransform": undefined,
      }
    }
  >
    Hello!
  </Text>
</View>
`;
